utils/swift_build_support/swift_build_support/targets.py

```diff
@@ -152,6 +152,34 @@ def __init__(self, platform, arch):
     def name(self):
         return "{}-{}".format(self.platform.name, self.arch)
 
+    def swift_flags(self, args, resource_dir_root=""):
+        flags = ''
+        if self.platform.name == 'android':
+            android_toolchain_path = '%s/toolchains/llvm/prebuilt/%s' % (
+                args.android_ndk, StdlibDeploymentTarget.host_target().name)
+
+            flags = '-target %s-unknown-linux-android%s ' % (self.arch,
+                args.android_api_level)
+            if resource_dir_root:
+                flags += '-resource-dir %s/lib/swift ' % (resource_dir_root)
+            else:
+                flags += '-resource-dir %s/swift-%s/lib/swift ' % (
+                    args.build_root, self.name)
+            flags += '-sdk %s/sysroot ' % (android_toolchain_path)
+            flags += '-tools-directory %s/bin' % (android_toolchain_path)
+
+        return flags
+
+    def cmake_options(self, args):
+        options = ''
+        if self.platform.name == 'android':
+            options = '-DCMAKE_SYSTEM_NAME=Android '
+            options += '-DCMAKE_SYSTEM_VERSION=%s ' % (args.android_api_level)
```

Comment:
Can you use the CMakeOptions() thing? The reason why is that I am going to be changing that to autogenerate Toolchain files and then I can just take what you already wrote here and it will just work.

(The plan is to make it so that we are always cross compiling with a toolchain).

Reply:
Also this is platform specific so I wonder if we can refactor this out onto the Android platform.

Reply:
I don't think so as CMakeOptions() is used to pass CMake flags in this Python `build-script`, whereas this is a string passed to the old bash `build-script-impl`. I suppose I could use CMakeOptions() to define these Android flags, then turn [that Python list](https://github.com/apple/swift/blob/6f4ef7cee24b33cc7be30c139ed0cd5b6e5e1358/utils/swift_build_support/swift_build_support/cmake.py#L31) into a string again before passing it to `build-script-impl`, but we could always do that later once you move everything to CMake toolchain files. Just let me know what you prefer.

As for refactoring it above, will do.

Reply:
I was thinking about this... I am fine with leaving it like this for now. But the refactoring I would like.